pytorch_lightning/callbacks/xla_stats_monitor.py

```diff
@@ -85,16 +85,18 @@ def on_train_epoch_start(self, trainer: "pl.Trainer", pl_module: "pl.LightningMo
         self._start_time = time.time()
 
     def on_train_epoch_end(self, trainer: "pl.Trainer", pl_module: "pl.LightningModule") -> None:
-        assert trainer.logger is not None
```

Comment:
I think assertion was there just to avoid mypy complaints. Error is already raised inside `on_train_start`.

Reply:
Yes the assert needs to be there. mypy cannot know the control flow and that we already raise in an earlier hook.

Reply:
but do we need to put the exception here again? I guess we don't do that elsewhere usually.

Reply:
`DeviceStatsMonitor` also has it in all hooks.

But this is more of a stylistic choice. Some people like no assertions, others prefer them since they are shorter.

If you feel strongly about this, you can open a PR converting the extra checks